### DISCUSSION TITLE
How can I format a textbox to show currency as the user types in H2O Wave?

### DISCUSSION THREAD

**Discussion by tvthanh0401** [h2oai/wave#1845] [Q&A]
I'm building an app using H2O Wave, and I want to format a textbox to show currency as the user types. For example, when the user types in the number 1200, the textbox should show $1,200.00. I'm able to format the value on the server-side using Python, but I want to update the textbox with the formatted value as the user types, without reloading the page.

**Comment by mturoci** [comment]
Hi @tvthanh0401. Maybe [textbox mask](https://wave.h2o.ai/docs/widgets/form/textbox/#with-mask) is what you are after?

**Reply by tvthanh0401** [reply]
oh, thank you so much, this is what I need.

**Reply by tvthanh0401** [reply]
Can we change the direction of mask? Currently, when I types in the number 1200, the textbox show 120,0

**Reply by mturoci** [reply]
Hm.. change of direction is not possible currently. Maybe try `mask='9,999.99'`? Or you can also use [spinbox](https://wave.h2o.ai/docs/widgets/form/spinbox/) and set `step=0.01` to allow for 2 decimal places.
